src/popover/index.js

```diff
@@ -65,49 +111,29 @@ class Popover extends Component {
       targetAttachment = 'middle right';
       break;
     }
-    return (
-      <span style={{position: 'relative', display: 'inline-block'}}>
-        {this._getElement()}
-        { this.props.isOpen &&
-          this._getArrowElement()
-        }
-        { this.props.isOpen &&
-          <TetherElement
-            target={this.refs.element}
-            options={{
-              attachment: `${attachment}`,
-              targetAttachment: `${targetAttachment}`,
-              constraints: [
-                {
-                  to: 'window',
-                  pin: ['left','right']
-                }
-              ]
-            }}
-          >
-            <PopoverContent
-              position={this.props.position}
-              shouldHaveBorder={this._shouldHaveBorder()}
-              style={this.props.style}
-              hasShadow={this.props.hasShadow} >
-              {this.props.children}
-            </PopoverContent>
-          </TetherElement>
-        }
-      </span>
-    );
-  }
-
-  _getElement = () => {
-    return React.cloneElement(this.props.element, { ref: 'element' });
-  }
 
-  _getArrowElement = () => {
     return (
-      <PopoverArrow
-        position={this.props.position}
-        shouldHaveBorder={this._shouldHaveBorder()}
-        style={this.props.style}/>
+      <TetherElement
+        target={this.refs.element}
+        options={{
+          attachment: `${attachment}`,
+          targetAttachment: `${targetAttachment}`,
+          constraints: [
```

Comment:
this is pretty hard to read, maybe define the constant at the top of render and pass it in?